packages/cactus-plugin-ledger-connector-iroha/src/main/typescript/plugin-ledger-connector-iroha.ts

```diff
@@ -140,9 +140,63 @@ export class PluginLedgerConnectorIroha
     this.log.info(`Shutting down ${this.className}...`);
   }
 
-  async registerWebServices(app: Express): Promise<IWebServiceEndpoint[]> {
+  async registerWebServices(
+    app: Express,
+    wsApi: SocketIoServer,
+  ): Promise<IWebServiceEndpoint[]> {
+    const { logLevel } = this.options;
     const webServices = await this.getOrCreateWebServices();
     await Promise.all(webServices.map((ws) => ws.registerExpress(app)));
+
+    wsApi.on("connection", (socket: SocketIoSocket) => {
+      let irohaSocketEndpoint: IrohaSocketIOEndpoint;
+
+      if (socket.id in this.socketSessionDictionary) {
+        this.log.debug(`Connected to existing socket session ID=${socket.id}`);
+        irohaSocketEndpoint = this.socketSessionDictionary[socket.id];
+      } else {
+        this.log.debug(`New Socket connected. ID=${socket.id}`);
+        irohaSocketEndpoint = new IrohaSocketIOEndpoint({ socket, logLevel });
+        this.socketSessionDictionary[socket.id] = irohaSocketEndpoint;
```

Comment:
@outSH Where in the code is this getting cleaned out of memory when the connection for the socket is closed?

Reply:
@petermetz It's not. I think handling socketio reconnections is more complex problem so I've removed this whole session storage completely. It has similar logic as watchBlock in other modules.

Reply:
@outSH Okay, that's fine by as well (end result is the same from the perspective of why I wrote the question). Thank you for addressing!